src/aspire/operators/polar_ft.py

```diff
@@ -116,15 +99,46 @@ def _evaluate_t(self, x):
             Fourier grid. This is an array of vectors whose first dimension
             corresponds to `x.shape[0]`, and last dimension equals `self.count`.
         """
-        nimgs = x.shape[0]
-
-        half_size = self.ntheta // 2
-
-        pf = nufft(x, self.freqs)
+        if x.dtype != self.dtype:
+            raise TypeError(
+                f"{self.__class__.__name__}::transform"
+                f" Inconsistent dtypes x: {x.dtype} self: {self.dtype}"
+            )
+
+        if not isinstance(x, Image):
+            raise TypeError(
+                f"{self.__class__.__name__}::transform"
+                f" passed numpy array instead of {Image}."
+            )
+        else:
+            x = x.asnumpy()
+
+        # Flatten stack
+        stack_shape = x.shape[: -self.ndim]
+        x = x.reshape(-1, *x.shape[-self.ndim :])
+
+        # We expect the Image `x` to be real in order to take advantage of the conjugate
+        # symmetry of the Fourier transform of a real valued image.
+        if not np.isreal(x).all():
+            raise TypeError(
+                f"The Image `x` must be real valued. Found dtype {x.dtype}."
+            )
+
+        resolution = x.shape[-1]
+
+        pf = nufft(x, self.freqs) / resolution**2
+
+        return pf.reshape(*stack_shape, self.ntheta // 2, self.nrad)
+
+    @staticmethod
+    def full(pf):
```

Comment:
Looks good. I see you didn't like my `pf = full(compact_pf)` naming idea eh 😇 😆 .

Reply:
I figured the "full" said enough 😇